File: SDKs/googleapis.com/my-business/v4/java/lib/src/main/java/openapisdk/models/shared/ReportLocationInsightsResponse.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * ReportLocationInsightsResponse
 * Response message for `Insights.ReportLocationInsights`.
**/
public class ReportLocationInsightsResponse {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("locationDrivingDirectionMetrics")
    public LocationDrivingDirectionMetrics[] locationDrivingDirectionMetrics;
    public ReportLocationInsightsResponse withLocationDrivingDirectionMetrics(LocationDrivingDirectionMetrics[] locationDrivingDirectionMetrics) {
        this.locationDrivingDirectionMetrics = locationDrivingDirectionMetrics;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("locationMetrics")
    public LocationMetrics[] locationMetrics;
    public ReportLocationInsightsResponse withLocationMetrics(LocationMetrics[] locationMetrics) {
        this.locationMetrics = locationMetrics;
        return this;
    }
}

File: SDKs/googleapis.com/my-business/v4/python/src/sdk/models/shared/reportlocationinsightsresponse.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import locationdrivingdirectionmetrics as shared_locationdrivingdirectionmetrics
from ..shared import locationmetrics as shared_locationmetrics


@dataclass_json
@dataclasses.dataclass
class ReportLocationInsightsResponse:
    r"""ReportLocationInsightsResponse
    Response message for `Insights.ReportLocationInsights`.
    """
    
    location_driving_direction_metrics: Optional[list[shared_locationdrivingdirectionmetrics.LocationDrivingDirectionMetrics]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('locationDrivingDirectionMetrics') }})
    location_metrics: Optional[list[shared_locationmetrics.LocationMetrics]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('locationMetrics') }})
    
